flask/app/variants.py

```diff
@@ -369,6 +519,21 @@ def summary(type: str):
             abort(500, "Unexpected error")
 
         agg_df = get_report_df(sql_df, type=type)
+
+        columns = request.args.get("columns", type=str)
+        if columns is not None:
+            columns = columns.split(",")
+            # filter out faulty columns
+            valid_columns = {col for col in agg_df.columns.values}
+            fixed_columns = [col for col in columns if col in valid_columns]
+            if len(fixed_columns) < len(columns):
+                app.logger.warn(
+                    "Ignoring invalid columns from request: {}".format(
+                        set(columns) ^ set(fixed_columns)
+                    )
+                )
+            agg_df = agg_df.loc[:, fixed_columns]
```

Comment:
the order the columns are specified in the query params will determine the order of the columns in the csv, overriding the order specified by `relevant_cols`. I think, if this is not accounted for in the front end, the order of the columns should be preserved in the report and possibly fixed, at least for chromosome, position, ref allele, and alt allele. thoughts?

Reply:
Should be fixed now. `fixed_columns` is now generated from `relevant_cols` while being compared to the requested columns and `valid_columns`.